src/Orchard.Web/Modules/Orchard.MediaLibrary/Controllers/LocalizedMediaController.cs

```diff
@@ -0,0 +1,153 @@
+            Logger = NullLogger.Instance;
+
+        }
+        public IOrchardServices Services { get; set; }
+        public Localizer T { get; set; }
+        public ILogger Logger { get; set; }
+        public ActionResult MediaItems(string folderPath, int skip = 0, int count = 0, string order = "created", string mediaType = "", string culture = "") {
+            if (String.IsNullOrWhiteSpace(folderPath)) {
+                folderPath = null;
+            }
+            if (!_mediaLibraryService.CheckMediaFolderPermission(Permissions.SelectMediaContent, folderPath)) {
+                Services.Notifier.Add(UI.Notify.NotifyType.Error, T("Cannot select media"));
+                var model = new MediaManagerMediaItemsViewModel {
+                    MediaItems = new List<MediaManagerMediaItemViewModel>(),
+                    MediaItemsCount = 0,
+                    FolderPath = folderPath
+                };
+
+                return View(model);
+            }
+
+            // Check permission
+            if (!_mediaLibraryService.CheckMediaFolderPermission(Permissions.SelectMediaContent, folderPath) && !_mediaLibraryService.CanManageMediaFolder(folderPath)) {
+                var model = new MediaManagerMediaItemsViewModel {
+                    MediaItems = new List<MediaManagerMediaItemViewModel>(),
+                    MediaItemsCount = 0,
+                    FolderPath = folderPath
+                };
+
+                return View(model);
+            }
+
+            IEnumerable<MediaPart> mediaParts;
+            var mediaPartsCount = 0;
+            if (culture == "") {
+                mediaParts = _mediaLibraryService.GetMediaContentItems(folderPath, skip, count, order, mediaType, VersionOptions.Latest);
+                mediaPartsCount = _mediaLibraryService.GetMediaContentItemsCount(folderPath, mediaType, VersionOptions.Latest);
+            }
+            else {
+                var cultureId = _cultureManager.GetCultureByName(culture).Id;
+                var query = BuildGetMediaContentItemsQuery(Services.ContentManager, folderPath, order: order, mediaType: mediaType, versionOptions: VersionOptions.Latest)
+                    .Join<LocalizationPartRecord>()
+                    .Where(x => x.CultureId == cultureId)
+                    .Join<MediaPartRecord>();
+                mediaParts = query
+                    .Slice(skip, count);
+                mediaPartsCount = query.Count();
+            }
+
+            var mediaItems = mediaParts.Select(x => new MediaManagerMediaItemViewModel {
+                MediaPart = x,
+                Shape = Services.ContentManager.BuildDisplay(x.ContentItem, "Thumbnail")
+            }).ToList();
+
+            var viewModel = new MediaManagerMediaItemsViewModel {
+                MediaItems = mediaItems,
+                MediaItemsCount = mediaPartsCount,
+                FolderPath = folderPath
+            };
+            return View(viewModel);
+        }
```

Comment:
I have replicated the logic within Orchard.Search, even if I don't like so much the results because filters from different features can't works together. Basically I have not introduced anything worse than it already was, so i ask you if you can merge it anyway. 
But we should consider to change the current approach: 
I don't know if it is possible but:
- the responsible of acting the filter should be the medialibrary module
- other modules can partecipate in building the where clause of the query (as per projections)
- every change in the filters form/folders navigation fires a medialibrary controller responsible of combining the queries and then exposing the result.
@sebastienros if you agree I will file a new issue describing the problem and proposing the above solution.